zookeeper-server/src/test/java/org/apache/zookeeper/test/EmptiedSnapshotRecoveryTest.java

```diff
@@ -143,6 +144,48 @@ public void testRestoreWithTrustedEmptySnapFiles() throws Exception {
         runTest(false, true);
     }
 
+    @Test
+    public void testRestoreWithTrustedEmptySnapFilesWhenFollowing() throws Exception {
+        QuorumUtil qu = new QuorumUtil(1);
+        try {
+            qu.startAll();
+            String connString = qu.getConnectionStringForServer(1);
+            try (ZooKeeper zk = new ZooKeeper(connString, CONNECTION_TIMEOUT, this)) {
+                for (int i = 0; i < N_TRANSACTIONS; i++) {
+                    zk.create("/node-" + i, new byte[0], Ids.OPEN_ACL_UNSAFE, CreateMode.PERSISTENT);
+                }
+            }
```

Comment:
This section creates transactions, ensuring a reasonable `xzid`, but might *not* result in a snapshot, as it doesn't lower the `snapCount` using `SyncRequestProcessor.setSnapCount`.  (The other tests do, and might run before.  Might be a good idea to be explicit if the goal of running `N_TRANSACTIONS` is to create one?)

This does *not* mean that the test is broken: a snapshot is present, but it is there because of the `initialize` file left behind by `ClientBase.createTmpDir`.

Reply:
The goal here is not to make the nodes write a snapshot. I just want to put some data in Zookeeper, since if Zookeeper is empty, it will create a snapshot file on boot, even without this change.

The behavior this PR changes is that if Zookeeper is booted with the trust empty property set to true, and the cluster is not empty, then the follower nodes will write a snapshot file if they don't already have one.

I have added a few lines to the test to assert that the data created here is still there once the followers come up, that seems like a healthy thing to check.

Reply:
I guess I could use another number than N_TRANSACTIONS if it is misleading?

Reply:
> The goal here is not to make the nodes write a snapshot due to hitting the snapcount. I just want to put some data in Zookeeper, since if Zookeeper is empty, it will create a snapshot file on boot, even without this change.

Okay.

(I understand the rest of the PR, and don't have any objections.  I was just wondering if the `N_TRANSACTIONS` was because you were trying to hit `snapCount`, like other tests in that file seem to be doing.)

Thank you for the clarification!

Reply:
Just to summarize what this test is intended to do:

It is supposed to set up a quorum with some data in it. It then deletes the snapshots on all follower nodes. This simulates what a low-traffic cluster being upgraded from 3.4 might look like. The cluster is restarted with ZOOKEEPER_SNAPSHOT_TRUST_EMPTY=true, and the test verifies that all nodes get a snapshot written. This will allow the operator to remove ZOOKEEPER_SNAPSHOT_TRUST_EMPTY again and reboot the cluster shortly after, easing migration.

Reply:
Right; we are on the same page :)

Would you mind rebasing this on top of the latest `master`, so that CI picks the new configuration and re-runs the test suite with a higher chance of success?

Reply:
Sure. Rebased and squashed. FWIW I ran the two failing tests locally, and they passed, so am guessing it's just flakiness.

Reply:
Also just a note on rebasing on top of master, since the behavior surprised me, and I thought I'd share: GH Actions' checkout action does not check out the PR branch, it instead creates a merge commit between the PR branch and the target branch, which it then runs the rest of the CI workflow on. So likely the previous run did run with the config from master.